server/controllers/vote.controller.js

```diff
@@ -121,28 +105,28 @@ function upvote(req, res, next) {
       }
 
       if (vote.active) {
-        post.score += incrementValue;
-        raccoon.liked(userIdString, postIdString);
+        entity.score += incrementValue;
+        req.liked = true;
       } else {
-        post.score -= incrementValue;
-        raccoon.unliked(userIdString, postIdString);
+        entity.score -= incrementValue;
+        req.unliked = true;
       }
-
-      return Bluebird.all([vote.save(), post.save()]);
-    }
+      promise = Bluebird.all([vote.save(), entity.save()]);
+    } else {
 
     const newvote = new Vote();
-    newvote.postId = post._id;
+    newvote.entityId = entity._id;
```

Comment:
We need to keep the postId until the business logic is abstracted.

Reply:
This makes sense, I considered this approach and it also has the added benefit of not breaking the tests.